pkg/controller/status/status.go

```diff
@@ -443,21 +467,9 @@ func (m *statusManager) syncState() bool {
 		}
 	}
 
-	if len(m.deployments)+len(m.daemonsets)+len(m.statefulsets)+len(m.cronjobs) > 0 {
```

Comment:
Oh haha, I see that you changed this to align with the other changes. That makes more sense.

Reply:
> That is effectively a "Has syncState() completed?" check

If that's what we want then an _explicit_ "has sync run" field feels more appropriate. This isn't easily understood by reading this, a dev wouldn't know easily that progressing and failing being false is a signal that this hasn't run. It also means that a change to this behaviour in the sync function breaks everything, so it's a bit of a fragile signal.

Reply:
> ideally the other functions wouldn't need to know about the internals of how syncState works to know that it's complete, but it is what it is.

Yep, this is what I was hinting at with this vague comment. Totally agree.